win32/src/threadpool_win32.c

```diff
@@ -86,20 +90,16 @@ static void internal_close(THREADPOOL_HANDLE threadpool)
         {
             break;
         }
-
         (void)WaitOnAddress(&threadpool->pending_api_calls, &current_pending_api_calls, sizeof(current_pending_api_calls), INFINITE);
     } while (1);
 
-    /* Codes_SRS_THREADPOOL_WIN32_01_030: [ threadpool_close shall wait for any executing callbacks by calling CloseThreadpoolCleanupGroupMembers, passing FALSE as fCancelPendingCallbacks. ]*/
-    CloseThreadpoolCleanupGroupMembers(threadpool->tp_cleanup_group, FALSE, NULL);
-
     /* Codes_SRS_THREADPOOL_WIN32_01_032: [ threadpool_close shall close the threadpool cleanup group by calling CloseThreadpoolCleanupGroup. ]*/
     CloseThreadpoolCleanupGroup(threadpool->tp_cleanup_group);
 
     /* Codes_SRS_THREADPOOL_WIN32_01_033: [ threadpool_close shall destroy the thread pool environment created in threadpool_open_async. ]*/
     DestroyThreadpoolEnvironment(&threadpool->tp_environment);
 
-    (void)InterlockedExchange(&threadpool->state, (LONG)THREADPOOL_WIN32_STATE_CLOSED);
+    (void)InterlockedExchange(&threadpool->state, THREADPOOL_WIN32_STATE_CLOSED);
     WakeByAddressSingle((PVOID)&threadpool->state);

```

Comment:
>WakeByAddressSingle [](http://example.com/codeflow?start=4&length=19)

How do we ensure that only 1 thread is waiting for this address? Why use "Single" instead of "All"?

Reply:
We are now waiting on the pending_api_calls, before the pending_api_calls was not waiting on the work function to execute, now we are, that is ensuring that we have completed all the work.

Reply:
OK, so the only line where this is waited is 180, and that's in "destroy" which by convention it is "just one destroy ever".